packages/amplify-graphql-relational-transformer/src/graphql-has-many-transformer.ts

```diff
@@ -147,7 +155,12 @@ export class HasManyTransformer extends TransformerPluginBase {
     const context = ctx as TransformerContextProvider;
 
     for (const config of this.directiveList) {
-      config.relatedTypeIndex = getRelatedTypeIndex(config, context, config.indexName);
+      const dbType = ctx.modelToDatasourceMap.get(getBaseType(config.field.type))?.dbType ?? DDB_DB_TYPE;
```

Comment:
I have moved this logic into an utility method into transformer core package in the PR I'm currently working to support hasOne and belongsTo directives so that this can reused across all relational directives.